### DISCUSSION TITLE
Name collision when renaming a field

### DISCUSSION THREAD

**Discussion by RenaudAubert** [prisma/prisma#19995] [Q&A] [answered]
Hi !

I encountered an error in my `schema.prisma` file and I was wondering if I was doing something forbidden or if it might be a bug?
```prisma
model Site {
  id            String @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
  technicalName String @map("name") @db.VarChar // Field "technicalName" is already defined on model "Site"
  name          String @map("display_name") @db.VarChar

  @@map("sites")
}
```

I have 2 fields:
- `name` that I want to rename `technicalName`
- `display_name` that I want to rename `name`

The problem here is that the prisma extension on VSCode is giving me an error saying stating
![image](https://github.com/prisma/prisma/assets/15106918/698d9f05-0994-423f-8a38-82f16810c548)

I tried reversing the 2 lines and the errordisappears. This is what makes me think it might be a bug but I want to be sure I'm not doing something wrong.

I'm using node `18.16.1`, with prisma `4.16.1` both as npm package and VSCode extension.

Cheers

**Comment by nurul3101** [comment, marked as answer]
Hello @RenaudAubert 👋 

In your example, when Prisma parses the `technicalName` field, it tries to map it to the database column `name`. However, in the next line, you're defining a new field called `name`. This causes a conflict because Prisma thinks that you're trying to define `name` twice (once as `technicalName` mapped to `name`, and then again as `name`).

To resolve this, you can simply change the order of the lines (which you did):

```prisma
model Site {
  id            String @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
  name          String @map("display_name") @db.VarChar
  technicalName String @map("name") @db.VarChar

  @@map("sites")
}
```

By defining the `name` field first (which maps to `display_name` in the database), and then defining the `technicalName` field (which maps to `name` in the database), Prisma can correctly interpret your intentions without conflict.

Hope this helps!

**Reply by RenaudAubert** [reply]
Hello @nurul3101

Thank you for replying so quickly. I confirm this is how I resolved the issue. 
I asked this question because I wasn't sure if it was normal behavior but thanks to you I now know it is indeed the correct behavior. 😄 

Cheers!

**Reply by nurul3101** [reply]
I am glad to hear that! :)
